app/controllers/concerns/omniauth_callback_behavior.rb

```diff
@@ -0,0 +1,61 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# mixin for okta callbacks,
+# note failure is configured in omniauth initializer
+module OmniauthCallbackBehavior
+  extend ActiveSupport::Concern
+  include AuthenticatesWithTwoFactor
+
+  def okta
+    log("begin user_type:#{user_scope.name}")
+
+    auth = request.env['omniauth.auth']
+    provider = auth.provider
+    user = OmniauthUserProvisioner.new.perform(auth: auth, user_scope: user_scope)
+
+    log("found #{user.class.name}##{user.id}")
+    if user.two_factor_enabled?
+      handle_2fa(user)
+    else
+      log('sign-in')
+      cookies.signed[:active_provider] = provider
+      handle_success(user)
+    end
+  end
+
+  def passthru
+    # Just send them to the home page for now instead
+    # of showing the ugly text messsage
+
+    redirect_to home_path
+  end
+
+  protected
+
+  def home_path
+    root_path
```

Comment:
This should be `user.my_root_path` for warehouse logins.

Reply:
Looking at it again, this was only used by passthru, which was a vestige of devis/omniauthable. I don't think it's needed, removed this method